pkg/virt-controller/watch/clone/clone_test.go

```diff
@@ -879,6 +891,29 @@ var _ = Describe("Clone", func() {
 				controller.Execute()
 			})
 
+			It("should generate patches from the vmsnapshotcontent, instead of the current VM", func() {
+				if sourceVM.Annotations == nil {
+					sourceVM.Annotations = make(map[string]string)
+				}
+				// add annotation to create dis-alignment between vm and vmsnapshotcontent
+				sourceVM.Annotations["new_annotation_matching_filter"] = "ok"
+				vmClone.Spec.AnnotationFilters = []string{"somekey/*"}
+				addVM(sourceVM)
+				addClone(vmClone)
+
+				client.Fake.PrependReactor("create", restoreResource, func(action testing.Action) (handled bool, ret runtime.Object, err error) {
```

Comment:
You can just check the restore or not? (after the `Execute` of course)

Reply:
You mean using the fakeclient?
If yes, I have a refactoring PR from reactor to fakeclient, and I would prefer to address this there.
Just to have similarity inside the file.
WDYT?
Thanks

Reply:
hm, I find it counter-productive and IMHO it could be more readable from now on.

Reply:
In the `BeforeEach` there is a reactor that expect that no request should arrive to the client that was not planned.
```
client.Fake.PrependReactor("*", "*", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
			Expect(action).To(BeNil())
			return true, nil, nil
		})
```
IOW for how it is structured right now, in order to get the vmrestore from the client I have to:
- prepend the vm restore creation anyway (otherwise the test fails due to a non planned request)
- prepend the vm restore get that I am going to perform to the client
- check the vm restore pathces.

I can, otherwise, include the refactoring in this PR, but this will increase the scope.
As you prefer. Please, let me know! Thanks